spec.md

```diff
@@ -289,6 +290,34 @@ encapsulated within the `data` attribute.
 * Constraints:
   * OPTIONAL
 
+# Minimum Supported Event Size
+
+In order to increase interoperability, all CloudEvent consumers SHOULD accept
+events that follow these rules:
+
+* The number of attributes does not exceed 100. Each index of a `Map` is
+  considered an attribute.
+* All attribute names are 20 characters or less long.
+* All indexes of `Maps` are 20 characters or less long.
+* All `Binary` attributes do not exceed a size of 1KB.
+* All `String` attributes (including String expressions like `URI-reference` and
+  `Timestamp`) do not exceed a size of 1KB.
+* `Any` attributes containing a `Binary` or a `String` follow the respective
+  rule above. The `data` attribute is excempt from this rule.
+* For a `Map`, the `Any`-typed values follow the rules above.
+* The total size of all attributes (including the `data` attribute) does not
+  exceed 40KB/64KB.
+
+The size of a `String` attribute is measured by encoding it in
+[UTF-8](https://tools.ietf.org/html/rfc3629).
+The size of an `Integer` attribute is 4 bytes.
```

Comment:
Why do we assume 4 bytes for `Integer`? That seems unlikely to be correct as an encoding estimate, and may not be correct as a length estimate (for example, for call stack traces from a running program, as a fraction of instruction executions).

Reply:
Note that `Integer` is defined as a 32-bit Integer, so 4 byte is the minimal encoding size for the value itself. Practically, the encoding may be bigger (e.g. in JSON).

Actually, I don't care much what the particular value is, but there should be some encoding-independent value to calculate the total size. Do you have a better proposal than 4 byte?

Reply:
A 32-bit integer can consume up to 10 bytes encoded in decimal. If we're estimating worst-case size, then 10 bytes might be a better estimate.

Reply:
I am not trying to estimating worst-case size (which in this case would probably be in JSON, where the Integer is represented as text...).

The size calculated for the event will most likely be lower than any wire formatting will be able to achieve (without compression). The goal isn't to predict the size of a message, but to have a way to validate the size of an event independently of the chosen encoding and transport format.